bot/main.py

```diff
@@ -194,13 +217,10 @@ async def on_guild_join(guild):
     welcome_message = "CAS-bot has joined this server"
 
     if server_config is None:
-        welcome_message = "Read the instructions to invite the bot in the project README"
-
-    for channel in guild.text_channels:
-        if channel.permissions_for(guild.me).send_messages:
-            await channel.send(welcome_message)
-            break
+        welcome_message = "This server is not authorized to work with CAS-bot. Read the instructions to invite the bot in the project README."
 
+    await get_first_channel_with_permission(guild).send(welcome_message)
```

Comment:
`get_first_channel_with_permission` can return `None` in the case that there exists no channel the bot is allowed to write to. Add appropriate try-except error handling